README.md
English
=======
Authors
-------
* _Hayato_ (Owner)
* Olivar Nax (Developer & Maintainer)
Features
--------
* Preview multiple items in the same window
* Preview items you can't equip
* Dye items you're currently previewing! (Only works with items from your Inventory atm)
* Preview FABKits. Only the base tier of an 'tiered' FABkit has any preview screenshots, e.g. Garden Tier 1 -> Works, Garden Tier 2 -> doesn't work
* Preview items on Auction House, from Quest Rewards, Questlog, Chatlinks, Schematics/Tech Tree Tab, Challenges and Greed Vs Need Roll Window
Supported Addons
----------------
* *ChatLog Addons*: ChatLog (Carbine), BetterChatLog, ChatFixed, ImprovedChatLog, FixedChatLog
* *QuestLog Addons*: QuestLog (Carbine), BetterQuestLog
* *Vendor Addons*: Vendor (Carbine), LilVendor
* *AuctionHouse Addons*: MarketplaceAuction (Carbine), EZAuction
* *ChallengeReward Addons*: ChallengeRewardPanel (Carbine)
* *Roll Addons*: NeedVsGreed (Carbine)
* *Dialog Addons*: Dialog (Carbine), ClassicQuestDialog, InstantQuestAccept, QuickDialog, UnitedDialogs, Ern's Speedy Quest
* *Techtree Addons*: Tradeskills (Carbine)
* *Schematics Addons*: TradeskillSchematics (Carbine)
How to use it
-------------
Ctrl + Right Click
To Do List
----------
* Add bank preview
Known Bugs
----------
* Switching between normal mode and light mode will reset its position to the last known position of the window in normal mode
* If you can't preview any new FABkit which was implemented with Content Drop 1 teleport to your house please and try again! (you have to do that just once)
German
======
Authors
-------
* _Hayato_ (Owner)
* Olivar Nax (Developer & Maintainer)
Features
--------
* Vorschau mehrerer Gegenstände in einem Vorschau-Fenster
* Vorschau von Gegenständen, die man nicht ausrüsten kann
* Färbe Items, die du momentan in der Vorschau anzeigen lässt! (Funktioniert aktuell nur mit Items aus dem Inventar)
* Vorschau von BAUsätzen. BAUsätze mit mehreren Rängen (Garten Rang 1, Rang 2 usw.) haben lediglich Screenshots für ihren Basisrang, also 1.
* Vorschau von Gegenständen im Auktionshaus, während man eine Questbelohnung auswählt bzw. Questlogbelohnungen, Chatlinks, Baupläne/Techbaum Tabs, Herausforderungsbelohnungen und Items, die man erwürfelt
Unterstützte Addons
-------------------
* *ChatLog Addons*: ChatLog (Carbine), BetterChatLog, ChatFixed, ImprovedChatLog, FixedChatLog
* *QuestLog Addons*: QuestLog (Carbine), BetterQuestLog
* *Vendor Addons*: Vendor (Carbine), LilVendor
* *AuctionHouse Addons*: MarketplaceAuction (Carbine), EZAuction
* *ChallengeReward Addons*: ChallengeRewardPanel (Carbine)
* *Roll Addons*: NeedVsGreed (Carbine)
* *Dialog Addons*: Dialog (Carbine), ClassicQuestDialog, InstantQuestAccept, QuickDialog, UnitedDialogs, Ern's Speedy Quest
* *Techtree Addons*: Tradeskills (Carbine)
* *Schematics Addons*: TradeskillSchematics (Carbine)
Wie man es verwendet
--------------------
Strg + Rechtsklick
To Do Liste
-----------
* Bankvorschau hinzufügen
Bekannte Probleme
-----------------
* Das Wechseln zwischen dem normalen und light Modus führt dazu, dass das Fenster sich zu der letzten bekannten Position des Fensters im Normalmodus zurück bewegt.
* Solltest du einen BAUsatz nicht in der Vorschau anzeigen lassen können, der neu mit dem Conten Drop 1 hinzugekommen ist, so teleportiere dich einmalig in dein Haus und versuche es erneut.

CHANGES.txt
lightweight tag 6a79225a279f7c8ba3eef954f813ccab118d11d9 v1.0.4
Author:	Arne De Herdt <arne.de.herdt@maxdome.de>
Date:	Thu Apr 16 14:13:11 2015 +0200

commit 6a79225a279f7c8ba3eef954f813ccab118d11d9
Author: Arne De Herdt <arne.de.herdt@maxdome.de>
Date:   Thu Apr 16 14:13:11 2015 +0200

    rewrote code, upped version

commit 4ba7a1d516ad003aae2d5574f8b438d03dc143c7
Author: Arne De Herdt <arne.de.herdt@maxdome.de>
Date:   Thu Apr 16 13:28:59 2015 +0200

    improved code readability



tables/FABKits.lua
ktFABkits = {
	[20746] = 2,
	[20747] = 1,
	[20735] = 81,
	[20579] = 33,
	[20675] = 15,
	[31368] = 24,
	[31370] = 84,
	[31371] = 80,
	[20547] = 22,
	[20576] = 31,
	[20589] = 16,
	[20614] = 26,
	[20573] = 11,
	[20707] = 29,
	[20662] = 91,
	[31372] = 64,
	[20548] = 27,
	[37182] = 38,
	[20571] = 79,
	[31373] = 68,
	[31374] = 93,
	[31375] = 55,
	[31382] = 82,
	[31377] = 92,
	[31383] = 25,
	[31386] = 45,
	[31376] = 61,
	[20546] = 21,
	[31385] = 35,
	[20667] = 18,
	[31357] = 74,
	[20552] = 71,
	[20616] = 42,
	[20549] = 43,
	[20554] = 4,
	[20686] = 36,
	[31351] = 88,
	[20695] = 62,
	[20652] = 63,
	[20666] = 28,
	[31354] = 96,
	[20619] = 56,
	[31350] = 77,
	[31355] = 90,
	[20654] = 41,
	[20647] = 23,
	[31360] = 76,
	[31359] = 69,
	[31356] = 87,
	[31358] = 72,
	[31363] = 75,
	[31365] = 99,
	[31369] = 70,
	[31361] = 73,
	[44805] = 102,
	[44806] = 103,
	[44807] = 105,
	[44808] = 104,
	[31394] = 94,
	[31387] = 20,
	[31388] = 83,
	[31390] = 65,
	[31364] = 97,
	[45433] = 100,
	[31379] = 60,
	[31366] = 89,
	[31367] = 98,
	[31380] = 78,
	[31384] = 58,
	[31378] = 57,
	[31389] = 95,
	[31397] = 85,
	[31381] = 32,
	[31391] = 86,
	[31392] = 30,
	[20698] = 66,
	[31393] = 67,
	[31395] = 59,
	[31398] = 44,
	[31399] = 40,
	[31401] = 19
}


lib/LibApolloFixes/LibApolloFixes-1.0.lua
local MAJOR, MINOR = "Lib:ApolloFixes-1.0", 1
-- Get a reference to the package information if any
local APkg = Apollo.GetPackage(MAJOR)
-- If there was an older version loaded we need to see if this is newer
if APkg and (APkg.nVersion > 0) then
	return -- no upgrades
end
-- Set a reference to the actual package or create an empty table
local Lib = APkg and APkg.tPackage or {}

-------------------------------------------------------------------------------
--- Local Variables
-------------------------------------------------------------------------------

-- Will contain an array of strings, the names of all Addons available loaded or not.
local tAddonList = {}

-- Use already stored original LoadForm in case of upgrade or get reference to the original LoadForm
Lib.fnOldLoadForm = Lib.fnOldLoadForm or Apollo.LoadForm
-- This is the number of elements in the table below it, used to determine if XML has been read in GetAddons()
local nNumObscured = 27
-- This list should be comprehensive as of 1.0.8.6745
-- Format is: ["FormName"] = "AddonName"
Lib.tObscuredAddons = Lib.tObscuredAddons or {
	["ImprovedSalvageForm"] = "ImprovedSalvage",
	["ItemPreviewForm"] = "ItemPreview",
	["TradeskillContainerForm"] = "TradeskillContainer",
	["TradeskillSchematicsForm"] = "TradeskillSchematics",
	["TradeskillTalentsForm"] = "TradeskillTalents",
	["WarpartyRegistrationForm"] = "WarpartyRegister",
	["WarpartyBattleForm"] = "WarpartyBattle",
	["TutorialTesterForm"] = "TutorialPrompts",
	["PathSoldierMissionMain"] = "PathSoldierMissions",
	["PathSettlerrMissionMain"] = "PathSettlerMissions",
	["PathScientistExperimentationForm"] = "PathScientistExperimentation",
	["PathScientistCustomizeForm"] = "PathScientistCustomize",
	["PowerMapRangeFinder"] = "PathExplorerMissions",
	["InteractionOnUnit"] = "HUDInteract",
	["HousingRemodelWindow"] = "HousingRemodel",
	["HousingLandscapeWindow"] = "HousingLandscape",
	["HousingListWindow"] = "HousingList",
	["DecorPreviewWindow"] = "DecorPreview",
	["PlugPreviewWindow"] = "PlugPreview",
	["MannequinWindow"] = "Mannequin",
	["HousingDatachronWindow"] = "HousingDatachron",
	["CircleRegistrationForm"] = "CircleRegistration",
	["GroupLeaderOptions"] = "GroupDisplayOptions",
	["FloaterPanel"] = "FloatTextPanel",
	["ChallengeLogForm"] = "ChallengeLog",
	["ChallengeRewardPanelForm"] = "ChallengeRewardPanel",
	["ArenaTeamRegistrationForm"] = "ArenaTeamRegister",
}

-- Use already stored original GetAddon in case of upgrade or get reference to the original GetAddon
Lib.fnOldGetAddon = Lib.fnOldGetAddon or Apollo.GetAddon
-- Populated once Addons are located within LoadForm
Lib.tFoundAddons = Lib.tFoundAddons or {}
-------------------------------------------------------------------------------
--- Local Functions
-------------------------------------------------------------------------------

local function GetAddons()
	-- Only parse XML once, if we have more than the number of obscured we had to have done this
	if #tAddonList > nNumObscured then
		return tAddonList
	end
	-- Addon Asset folders contain the full path, we parse off everything before the 1st occurance of Addons (regardless of capitalization)
	local strWildstarDir = string.match(Apollo.GetAssetFolder(), "(.-)[\\/][Aa][Dd][Dd][Oo][Nn][Ss]")
	-- Read in the Addons.xml file which is generated by Carbine and stores load states among other things
	local tAddonXML = XmlDoc.CreateFromFile(strWildstarDir.."\\Addons.xml"):ToTable()
	for k,v in pairs(tAddonXML) do
		-- We only care about the <Addon> tags
		if v.__XmlNode == "Addon" then
			if v.Carbine == "1" then -- Carbine Addons are the same name as thier folder (Apparently its a string 1 not a number 1)
				table.insert(tAddonList, v.Folder)
			else  -- User addons can have different folder/addon names so we have to parse their toc
				local xmlTOC = XmlDoc.CreateFromFile(strWildstarDir.."\\Addons\\"..v.Folder.."\\toc.xml")
				if xmlTOC then
					local tTocTable = xmlTOC:ToTable()
					table.insert(tAddonList, tTocTable.Name)
				end
			end
		end
	end
	return tAddonList
end

local function GetAddon(strAddonName)
	-- If this is one we found, return that
	if Lib.tFoundAddons[strAddonName] then
		return Lib.tFoundAddons[strAddonName]
	else
		-- Otherwise return the normal result
		return Lib.fnOldGetAddon(strAddonName)
	end
end

-- Gets all locals at a specified number of calls previous
local function DebugLocals(nLevel)
	local tVars, nIdx = {}, 1
	while true do
		local ln, lv = debug.getlocal(nLevel, nIdx)
		if ln ~= nil then
			tVars[ln] = lv
		else
			break
		end
		nIdx = nIdx + 1
	end
	return tVars
end

local function HookedLoadForm(...)
	-- Arg 2 is the Form Name
	local strForm = select(2, ...)
	if Lib.tObscuredAddons[strForm] then
		-- Pull local variables from 3 levels up
		local tDebugLocals = DebugLocals(3)
		-- Self at this point is the addon we are looking for so save that!
		Lib.tFoundAddons[Lib.tObscuredAddons[strForm]] = tDebugLocals.self
		-- Since we can now lookup this addon, lets add it to GetAddons()
		table.insert(tAddonList, Lib.tObscuredAddons[strForm])
		-- Found it so stop looking for this addon
		Lib.tObscuredAddons[strForm] = nil

		-- If this was the last thing we're looking for remove the hook
		if not next(Lib.tObscuredAddons) then
			Apollo.LoadForm = Lib.fnOldLoadForm
		end
	end
	-- Return the saved functions result
	return Lib.fnOldLoadForm(...)
end
Apollo.LoadForm = HookedLoadForm

function Lib:OnLoad()
	-- If Carbine ever adds GetAddons to Apollo lets not use ours.
	Apollo.GetAddons = Apollo.GetAddons or GetAddons
	Apollo.GetAddon = GetAddon
end

Apollo.RegisterPackage(Lib, MAJOR, MINOR, {})


tables/Mounts.lua
tMounts =
{	[1] =	{
			["Item_ID"] = 22123,
			["Item_Name"] = "Warpig Mount",
			["Mount_ID"] = 22297,
			["Hoverboard_ID"] = 0,
			},
	[2] =	{
			["Item_ID"] = 19183,
			["Item_Name"] = "Flux Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 6976,
			},
	[3] =	{
			["Item_ID"] = 29002,
			["Item_Name"] = "Uniblade Mount",
			["Mount_ID"] = 48178,
			["Hoverboard_ID"] = 0,
			},
	[4] =	{
			["Item_ID"] = 32904,
			["Item_Name"] = "Orbitron Mount",
			["Mount_ID"] = 52319,
			["Hoverboard_ID"] = 0,
			},
	[5] =	{
			["Item_ID"] = 37109,
			["Item_Name"] = "Velocirex Mount",
			["Mount_ID"] = 54261,
			["Hoverboard_ID"] = 0,
			},
	[6] =	{
			["Item_ID"] = 41487,
			["Item_Name"] = "Hellion Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7129,
			},
	[7] =	{
			["Item_ID"] = 41491,
			["Item_Name"] = "Rosie Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7133,
			},
	[8] =	{
			["Item_ID"] = 41490,
			["Item_Name"] = "Vortex Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7132,
			},
	[9] =	{
			["Item_ID"] = 41492,
			["Item_Name"] = "Cyclone Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7134,
			},
	[10] =	{
			["Item_ID"] = 41484,
			["Item_Name"] = "Spike Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7127,
			},
	[11] =	{
			["Item_ID"] = 41486,
			["Item_Name"] = "Go-Go Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7128,
			},
	[12] =	{
			["Item_ID"] = 38766,
			["Item_Name"] = "Woolie Mount",
			["Mount_ID"] = 56827,
			["Hoverboard_ID"] = 0,
			},
	[13] =	{
			["Item_ID"] = 44168,
			["Item_Name"] = "Savage Warpig Mount",
			["Mount_ID"] = 61428,
			["Hoverboard_ID"] = 0,
			},
	[14] =	{
			["Item_ID"] = 44113,
			["Item_Name"] = "Alien Velocirex Mount",
			["Mount_ID"] = 61390,
			["Hoverboard_ID"] = 0,
			},
	[15] =	{
			["Item_ID"] = 12424,
			["Item_Name"] = "Yockum Warboar",
			["Mount_ID"] = 17149,
			["Hoverboard_ID"] = 0,
			},
	[16] =	{
			["Item_ID"] = 44114,
			["Item_Name"] = "Marauder Orbitron Mount",
			["Mount_ID"] = 61391,
			["Hoverboard_ID"] = 0,
			},
	[17] =	{
			["Item_ID"] = 44436,
			["Item_Name"] = "Dreg Trask Mount",
			["Mount_ID"] = 61762,
			["Hoverboard_ID"] = 0,
			},
	[18] =	{
			["Item_ID"] = 44116,
			["Item_Name"] = "Hot Rod Uniblade Mount",
			["Mount_ID"] = 61393,
			["Hoverboard_ID"] = 0,
			},
	[19] =	{
			["Item_ID"] = 41024,
			["Item_Name"] = "Plasmatic Hyperboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7153,
			},
	[20] =	{
			["Item_ID"] = 44456,
			["Item_Name"] = "Rally Grinder Mount",
			["Mount_ID"] = 61833,
			["Hoverboard_ID"] = 0,
			},
	[21] =	{
			["Item_ID"] = 44837,
			["Item_Name"] = "Strain Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7294,
			},
	[22] =	{
			["Item_ID"] = 44455,
			["Item_Name"] = "War Woolie Mount",
			["Mount_ID"] = 61823,
			["Hoverboard_ID"] = 0,
			},
	[23] =	{
			["Item_ID"] = 44115,
			["Item_Name"] = "Run-Down Uniblade Mount",
			["Mount_ID"] = 61392,
			["Hoverboard_ID"] = 0,
			},
	[24] =	{
			["Item_ID"] = 41489,
			["Item_Name"] = "Manta Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7131,
			},
	[25] =	{
			["Item_ID"] = 41488,
			["Item_Name"] = "Bulldog Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7130,
			},
	[26] =	{
			["Item_ID"] = 41485,
			["Item_Name"] = "The Beast Hoverboard Mount",
			["Mount_ID"] = 62426,
			["Hoverboard_ID"] = 7126,
			},
	[27] =	{
			["Item_ID"] = 13130,
			["Item_Name"] = "Trask Mount",
			["Mount_ID"] = 5212,
			["Hoverboard_ID"] = 0,
			},
	[28] =	{
			["Item_ID"] = 5923,
			["Item_Name"] = "Equivar Mount",
			["Mount_ID"] = 8725,
			["Hoverboard_ID"] = 0,
			},
	[29] =	{
			["Item_ID"] = 4101,
			["Item_Name"] = "Grinder Mount",
			["Mount_ID"] = 11085,
			["Hoverboard_ID"] = 0,
			},
	[30] =	{
			["Item_ID"] = 44385,
			["Item_Name"] = "Luminous Equivar Mount",
			["Mount_ID"] = 61693,
			["Hoverboard_ID"] = 0,
			},
}
